modules/aws-codebuild-step/main.tf

```diff
@@ -0,0 +1,110 @@
+/**
+ * # AWS codebuild step
+ * AWS codebuild step is very simple module for building project using just one object in terraform.
+ * This module creates roles and codebuild project.
+ */
+
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+resource "aws_iam_role" "role" {
+  name = "${var.project_name}-role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "codebuild.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+data "aws_iam_policy_document" "task_role_assume" {
+  statement {
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents"
+    ]
+    effect = "Allow"
+    resources = [
+      "arn:aws:logs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:log-group:/aws/codebuild/${var.project_name}",
+      "arn:aws:logs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:log-group:/aws/codebuild/${var.project_name}:*"
+    ]
+  }
+
+  statement {
+    effect    = "Allow"
+    resources = ["${var.pipeline_s3_arn}/*"]
+    actions = [
+      "s3:PutObject",
+      "s3:GetObject",
+      "s3:GetObjectVersion",
+      "s3:GetBucketAcl",
+      "s3:GetBucketLocation"
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "codebuild:CreateReportGroup",
+      "codebuild:CreateReport",
+      "codebuild:UpdateReport",
+      "codebuild:BatchPutTestCases",
+      "codebuild:BatchPutCodeCoverages"
+    ]
+    resources = [
+      "arn:aws:codebuild:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:report-group/${var.project_name}-*"
+    ]
+  }
+}
+
+
+resource "aws_iam_role_policy" "role_policy" {
+  role   = aws_iam_role.role.name
+  policy = data.aws_iam_policy_document.task_role_assume.json
+}
+
+resource "aws_codebuild_project" "role" {
+  name          = var.project_name
+  description   = var.description
+  build_timeout = var.build_timeout
+  service_role  = aws_iam_role.role.arn
+
+  artifacts {
+    name = var.project_name
+    type = "CODEPIPELINE"
+  }
+
+  environment {
+    compute_type                = "BUILD_GENERAL1_SMALL"
```

Comment:
Would it not be better to make it a variable?

Reply:
Agree - you may need to have here more capacity.